src/encrypt.rs

```diff
@@ -103,3 +105,91 @@ pub(crate) fn encrypt_chunk(content: Bytes, pki: (Pad, Key, Iv)) -> Result<Bytes
     let encrypted = encryption::encrypt(Bytes::from(compressed), &key, &iv)?;
     Ok(xor(&encrypted, &pad))
 }
+
+/// Encrypt chunks in a streaming fashion, processing them in the correct order to satisfy the
+/// encryption requirements. Each chunk is encrypted using the hashes of two other chunks:
+/// - For chunk 0: Uses hashes of the last two chunks
+/// - For chunk 1: Uses hash of chunk 0 and the last chunk
+/// - For chunks 2+: Uses hashes of the previous two chunks
+pub(crate) fn encrypt_stream(
+    chunks: Vec<RawChunk>,
+) -> Result<(DataMap, Vec<EncryptedChunk>)> {
+    // Create a sorted vector of all hashes
```

Comment:
The memory impact of this one for `HUGE` files is `HUGE x 2` as `chunks` stays in memory all run time and `encrypted_chunks` as well. 
I don't understand how this is streaming?

Reply:
Nice catch, this was in another commit. I have added it in the code now and we actually do save chunks to the store as we process them and that should fix this one, but great catch